x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/openshiftai/action/OpenShiftAiActionCreatorTests.java

```diff
@@ -578,44 +572,136 @@ public void testExecute_TruncatesInputBeforeSending() throws IOException {
+                        {
+                            "index": 0,
+                            "document": {
+                                "text": "awdawdawda"
+                            },
+                            "relevance_score": 0.4921875
+                        }
+                    ]
+                }
+                """;
+            webServer.enqueue(new MockResponse().setResponseCode(200).setBody(responseJson));
+
+            var model = OpenShiftAiRerankModelTests.createModel(getUrl(webServer), API_KEY_VALUE, MODEL_VALUE);
+            var actionCreator = new OpenShiftAiActionCreator(
+                sender,
+                new ServiceComponents(threadPool, mockThrottlerManager(), NO_RETRY_SETTINGS, TruncatorTests.createTruncator())
+            );
+            var action = actionCreator.create(model, null);
+
+            PlainActionFuture<InferenceServiceResults> listener = new PlainActionFuture<>();
+            action.execute(
+                new QueryAndDocsInputs(QUERY_VALUE, DOCUMENTS_VALUE, null, null, false),
+                InferenceAction.Request.DEFAULT_TIMEOUT,
+                listener
+            );
+
+            var result = listener.actionGet(ESTestCase.TEST_REQUEST_TIMEOUT);
+            assertThat(result.asMap(), is(buildExpectationRerank(RERANK_EXPECTATIONS_WITH_TEXT_TWO_RESULTS)));
+        }
+        assertRerankActionCreator(DOCUMENTS_VALUE, 2, true);
+    }
+
+    public void testCreate_OpenShiftAiRerankModel_WithOverriddenTaskSettings() throws IOException {
+        var senderFactory = HttpRequestSenderTests.createSenderFactory(threadPool, clientManager, NO_RETRY_SETTINGS);
+
+        List<String> documents = DOCUMENTS_VALUE;
+        try (var sender = createSender(senderFactory)) {
+            sender.startSynchronously();
+
+            String responseJson = """
+                {
+                    "id": "rerank-d300256dd02b4c63b8a2bc34dcdad845",
+                    "model": "bge-reranker-v2-m3",
+                    "usage": {
+                        "total_tokens": 10
+                    },
+                    "results": [
+                        {
+                            "index": 0,
+                            "relevance_score": 0.4921875
+                        }
+                    ]
+                }
+                """;
+            webServer.enqueue(new MockResponse().setResponseCode(200).setBody(responseJson));
+
+            var model = OpenShiftAiRerankModelTests.createModel(getUrl(webServer), API_KEY_VALUE, MODEL_VALUE);
+            var actionCreator = new OpenShiftAiActionCreator(
+                sender,
+                new ServiceComponents(threadPool, mockThrottlerManager(), NO_RETRY_SETTINGS, TruncatorTests.createTruncator())
+            );
+            var action = actionCreator.create(
+                model,
+                new HashMap<>(Map.of(OpenShiftAiRerankTaskSettings.RETURN_DOCUMENTS, false, OpenShiftAiRerankTaskSettings.TOP_N, 1))
```

Comment:
Could the values of `false` and `1` be extracted to local variables here so they can be reused on line 698?